app/src/main/java/com/example/una/FirestoreClient.java

```diff
@@ -32,10 +32,35 @@
     private FirebaseUser user;
     private final String TAG = "FirestoreClient";
     private String userName;
+    private boolean isDonor;
 
     public FirestoreClient() {
         user = FirebaseAuth.getInstance().getCurrentUser();
-        getCurrentUserName();
+        if (user != null) {
```

Comment:
I believe @alexandrashaw10  already has something that can differentiate if the user is a donor or a charity